src/core/plugin/luapi_application.h

```diff
@@ -1976,6 +1977,135 @@ static int applib_openFile(lua_State* L) {
+
+        int heightParam = luaL_optinteger(L, -3, -1);
+        if (heightParam <= 0 && heightParam != -1)
+            return luaL_error(L, "Invalid height given, must be positive integer or -1 to deactivate manual setting.");
+
+        int widthParam = luaL_optinteger(L, -4, -1);
+        if (widthParam <= 0 && widthParam != -1)
+            return luaL_error(L, "Invalid width given, must be positive integer or -1 to deactivate manual setting.");
+
+        double scale = luaL_optnumber(L, -2, 1);
+        if (scale <= 0)
+            return luaL_error(L, "Invalid scale given, must be a positive number.");
+
+        bool aspectRatio{true};
+        if (lua_isnil(L, -1)) {
+            // map nil to true (not typical for lua) so that the default value is true
+            aspectRatio = true;
+        } else {
+            // use the typical lua version of booleans (everything different than false (any set value) is true)
+            aspectRatio = lua_toboolean(L, -1);
+        }
+
+        const char* path = luaL_checkstring(L, -7);
+        if (!path)
+            return luaL_error(L, "no 'path' parameter was provided.");
+
+        xoj::util::GObjectSPtr<GFile> file(g_file_new_for_path(path), xoj::util::adopt);
+        if (!g_file_query_exists(file.get(), NULL)) {
+            lua_pushstring(L, "Error: file does not exist.");  // soft error
+            return 1;
+        }
+
+        XojPageView* pv = control->getWindow()->getXournal()->getViewFor(control->getCurrentPageNo());
+        ImageHandler imgHandler(control, pv);
+        auto [succ, img, width, height] = imgHandler.createImage(file.get(), x, y);
+        if (!succ) {
+            lua_pushstring(L, "Error: creating the image failed.");  // soft error
+            return 1;
+        }
+
+        if (widthParam != -1 && heightParam != -1) {
+            width = widthParam;
+            height = heightParam;
+        } else if (widthParam != -1 && heightParam == -1) {
+            if (aspectRatio) {
+                heightParam = static_cast<int>(static_cast<double>(height) / static_cast<double>(width) * widthParam);
+                height = heightParam;
+            }
+            width = widthParam;
+        } else if (heightParam != -1 && widthParam == -1) {
+            if (aspectRatio) {
+                widthParam = static_cast<int>(static_cast<double>(width) / static_cast<double>(height) * heightParam);
+                width = widthParam;
+            }
+            height = heightParam;
+        }
+
+        width = static_cast<int>(width * scale);
+        height = static_cast<int>(height * scale);
+
+
+        img->setWidth(width);
+        img->setHeight(height);
+
+        succ = imgHandler.addImageToDocument(img);
+        if (!succ) {
+            lua_pushstring(L, "Error: Inserting the image failed.");  // soft error
+            return 1;
+        }
+
+        lua_pop(L, 7);  // pop the params we fetched from the param-table from the stack
```

Comment:
That's not the right number, I think.

Reply:
I think `7` was right, because there are 7 params per image parameters and the way I was accessing the different image parameter tables did work more like a FIFO (working its way bottom up in the stack, not top down) so I was using the index to iterate over the tables.
But the number changed anyhow as the argument passing was changed.